New CLI option for 'move-after-upload'

samwilson opened the pull request:
This adds the --move (or -m) option for providing a destination
directory for uploaded files to be moved to after upload. It
removes this option from the config file (and so is a non-BC
change).

If --move is not specified, the current default behaviour of not
moving files is retained.

samwilson committed:
New CLI option for 'move-after-upload'

This adds the --move (or -m) option for providing a destination
directory for uploaded files to be moved to after upload. It
removes this option from the config file (and so is a non-BC
change).

If --move is not specified, the current default behaviour of not
moving files is retained.

TheFox reviewed (changes requested):
Successfully tested. You can also remove the `upload` array on AuthCommand where the config file will be created. I like your detailed commit messages.

samwilson commented:
Thanks for checking this out (I'm back from summer holidays now and wading through any number of emails!). I've fixed up the auth command, and rebased. Should be good to go now, but check again just to make sure. :-) Thanks!
